dataprep/clean/clean_text.py

```diff
@@ -0,0 +1,441 @@
+"""
+Clean a DataFrame column containing text data.
+"""
+import re
+import string
+from functools import partial, update_wrapper
+from typing import Any, Callable, Dict, List, Optional, Set, Union
+from unicodedata import normalize
+
+import dask.dataframe as dd
+import numpy as np
+import pandas as pd
+
+from ..assets.english_stopwords import english_stopwords
+from .utils import NULL_VALUES, to_dask
+
+REGEX_BRACKETS = {
+    "angle": re.compile(r"(\<)[^<>]*(\>)"),
+    "curly": re.compile(r"(\{)[^{}]*(\})"),
+    "round": re.compile(r"(\()[^()]*(\))"),
+    "square": re.compile(r"(\[)[^\[\]]*(\])"),
+}
+REGEX_DIGITS = re.compile(r"\d+")
+REGEX_DIGITS_BLOCK = re.compile(r"\b\d+\b")
+REGEX_HTML = re.compile(r"<[A-Za-z/][^>]*>|&(?:[a-z0-9]+|#[0-9]{1,6}|#x[0-9a-f]{1,6});")
+REGEX_PUNCTUATION = re.compile(r"[%s]" % re.escape(string.punctuation))
+REGEX_URL = re.compile(r"(?:https?://|www\.)\S+")
```

Comment:
Using the URL regex from `clean_url()` is about twice as slow. Since we don't need to validate the URL, I thought we could simplify the regex here. Compared to texthero, we match strings starting with "http" and "url" (they only match "http").

Reply:
Good to know, thanks for trying both regexes. I agree with your thought process - the simpler regex is sufficient.